### DISCUSSION TITLE
Welcome to scPrisma Discussions!

### DISCUSSION THREAD

**Discussion by JonathanKarin** [nitzanlab/scPrisma#2] [Announcements]
Welcome to the scPrisma community! We're thrilled to have you all here and can't wait to see the discussions and ideas that will come out of this group.💪

We encourage you to participate in the following ways:

- Ask any questions you may have about using scPrisma.
- Report any bugs you encounter while using the package.
- Share new usage ideas or extensions for scPrisma.
- Engage in any other discussions related to scPrisma.

Let's jump in and make the most of this community!

**Comment by AndreaSalati** [comment]
Hi! great work! I was wandering, did you implement the linear reconstruction? Meaning the ordering of the rows for linear topologies?
(the equivalent of algorithms.reconstruction_cyclic() )?

**Reply by JonathanKarin** [reply]
We wrote about it in a few words in the discussion section, our reconstruction problem is not a convex optimization problem and has a challenging runtime complexity due to the Projection onto the Birkhoff Polytope. Therefore, for linear reconstruction (which is a very simple and known task), I recommend using [sc.tl.dpt](https://scanpy.readthedocs.io/en/stable/generated/scanpy.tl.dpt.html) (or any other linear pseudotime inference algorithm). If you prefer a topology base algorithm you can also use [NovoSparc](https://www.nature.com/articles/s41596-021-00573-7).

Why does our reconstruction algorithm work well in the cyclic case?
The cyclic task is a more challenging task and in this particular case, the permutation problem has 2n optimal solutions (any possible shift), and the analogous problem for 'dpt' in the circular case is the traveling salesman problem, which is NP-hard.

Jonathan
